ui/src/literature/components/DOILinkAction.jsx

```diff
@@ -1,30 +1,66 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { List } from 'immutable';
+import { Button, Menu } from 'antd';
 
 import ListItemAction from '../../common/components/ListItemAction';
-import EventTracker from '../../common/components/EventTracker';
+import DropdownMenu from '../../common/components/DropdownMenu';
+import IconText from '../../common/components/IconText';
+import DOILink from './DOILink';
+import DOIMaterial from './DOIMaterial';
 
 class DOILinkAction extends Component {
+  renderDoiDropdownItem() {
+    const { dois } = this.props;
+    return dois.map(doi => (
+      <Menu.Item key={doi.get('value')}>
+        <DOILink doi={doi.get('value')}>
+          {doi.get('value')}
+          <DOIMaterial material={doi.get('material')} />
+        </DOILink>
+      </Menu.Item>
+    ));
+  }
+
+  renderDoiDropdown(text, type) {
+    return (
+      <DropdownMenu
+        title={
+          <Button>
+            <IconText text={text} type={type} />
+          </Button>
+        }
+      >
+        {this.renderDoiDropdownItem()}
+      </DropdownMenu>
+    );
+  }
+
+  renderDoiLink(text, type) {
+    const { dois } = this.props;
+    return (
+      <DOILink doi={dois.get(0).get('value')}>
+        <IconText text={text} type={type} />
+      </DOILink>
+    );
+  }
+
   render() {
-    const { doi } = this.props;
-    const href = `//doi.org/${doi}`;
+    const { dois } = this.props;
+    const text = 'DOI';
```

Comment:
no need to pass those, just have `static renderIconTextForDOI()` that returns  `<IconText text="DOI" type="link"/>` and use in both places.